gradle/common-dependencies.gradle

```diff
@@ -30,6 +30,8 @@ dependencies {
   testImplementation 'org.junit.jupiter:junit-jupiter-params'
   testRuntimeOnly 'org.junit.vintage:junit-vintage-engine'
 
+  testImplementation 'org.assertj:assertj-core'
```

Comment:
Why do we need a new assertion dependency ? jupiter already exposes assertEquals

Reply:
I think the biggest difference is that `assertj` provides more context in assertions so that's what I typically use. Another benefit is the enhanced dev experience in tests. note: other projects in this repo are already using this, just not this one in particular.

ref: https://medium.com/nerd-for-tech/junit-vs-assertj-choosing-the-right-testing-framework-for-your-java-project-84d2664736dc